src/opera/parser/tosca/csar.py

```diff
@@ -1,19 +1,92 @@
+import shutil
+import yaml
+
 from pathlib import Path
 from tempfile import TemporaryDirectory
 from zipfile import ZipFile
 
-import yaml
 from opera.error import ParseError
 
 
 class CloudServiceArchive:
-    def __init__(self, csar_name, csar_folder_path):
+    def __init__(self, csar_name, csar_folder_path=None):
         self._csar_name = csar_name
         self._csar_folder_path = csar_folder_path
         self._tosca_meta = None
         self._root_yaml_template = None
         self._metadata = None
 
+    def package_csar(self, output, service_template=None, csar_format="zip"):
+        try:
+            root_yaml_files = []
+            root_yaml_files.extend(Path(self._csar_name).glob('*.yaml'))
+            root_yaml_files.extend(Path(self._csar_name).glob('*.yml'))
+
+            if len(root_yaml_files) != 1:
+                raise ParseError("There should be one root level yaml "
+                                 "file in the root of the CSAR: {}."
+                                 .format(list(map(str, root_yaml_files))),
+                                 self)
+
+            if not service_template:
+                service_template = root_yaml_files[0].name
+            else:
+                if not Path(self._csar_name).joinpath(
+                        service_template).exists():
+                    raise ParseError('The supplied TOSCA service template '
+                                     'file "{}" does not exist in folder '
+                                     '"{}".'.format(service_template,
+                                                    self._csar_name), self)
+
+            meta_file_path = Path(
+                self._csar_name) / "TOSCA-Metadata" / "TOSCA.meta"
+            if meta_file_path.exists():
```

Comment:
If you get `--service-template` specified and a meta file with it set exists, you silently ignore that, right? I don't know whether you should error, warn, or just silently change the value in the meta file.

Reply:
Aha, do you mean the case when I would specify TOSCA service template path with `--service-template` to one file and then in the `TOSCA.meta` (within `Entry-Definitions`) it would be set to another different file path. Yes, this could be a problem as here I only take the path in `TOSCA.meta` and forget about the first one. This can be resolved by adding a check if the two entries match. If they don't I believe that an error should be thrown.

Reply:
A said I've added check for that:

```python
                if service_template and template_entry != service_template:
                    raise ParseError('The file entry "{}" defined within '
                                     '"Entry-Definitions" in '
                                     '"TOSCA-Metadata/TOSCA.meta" does not '
                                     'match with the file name "{}" supplied '
                                     'in service_template CLI argument.'.
                                     format(template_entry, service_template),
                                     self)
```